CHANGELOG.md

```diff
@@ -1,24 +1,222 @@
+{ type: "int[]" }                             // Bad
+"int[]"                                       // Good
+{ type: "list", objectType: "int" }           // Good
+
+{ type: "int?" }                              // Bad
+"int?"                                        // Good
+{ type: "int", optional: true }               // Good
+
+// Specifying object types
+{ type: "SomeType" }                          // Bad
+"SomeType"                                    // Good
+{ type: "object", objectType: "SomeType" }    // Good
+
+{ type: "object[]", objectType: "SomeType" }  // Bad
+"SomeType[]"                                  // Good
+{ type: "list", objectType: "SomeType" }      // Good
+
+{ type: "linkingObjects", objectType: "SomeType", property: "someProperty" } // Good
+```
+
+* To prevent modifying end-users' class-based model classes, we’re now wrapping class-based models in our own model. Objects will still pass `instanceof SomeClass` checks, however, code which is directly using prototype or constructor comparisons will fail:
+  
+```javascript
+Object.getPrototypeOf(object) == CustomObject.prototype // No longer works
+object.constructor == CustomObject // No longer works
+```
+
+* Symbols used to be accepted as keys in a dictionary, where they were coerced to strings prior to performing lookup. This was undocumented behaviour that makes little sense in practice (and arguably defeats the main purpose of the JS `Symbol` type). In the new SDK, using a Symbol as a key in a dictionary will throw.
+* As a part of migrating to NAPI (since ~ v6), we saw no performant way to support getting property names of a Realm.Object via the standard Object.keys(obj). As a side-effect we stopped supporting the object spread operator `{...obj}` and introduced `Realm.Object#keys()`, `Realm.Object#entries()` and `Realm.Object#toJSON()` methods were introduced as a workaround. The new SDK wraps its accessor objects in a Proxy trapping the ownKeys operation which enables calls to the standard `Object.keys(obj)` and the spread operator `{...obj}` to work correctly, with minimal performance impact on normal accesses. Therefore, we are deprecating the APIs with the @deprecation annotation and a `console.warn` when calling RealmObject#keys() and RealmObject#entries(). RealmObject#toJSON still serves the purpose of producing a circularly referencing object graph. We would love the community's feedback on this!
+* The [push service](https://www.mongodb.com/docs/atlas/app-services/reference/push-notifications/) has already been deprecated on the Atlas server. We've deprecated this on v11 and removed it from v12.
+* We’ve decided to remove numeric indexing and “array methods” from the SubscriptionSet, since (a) the team saw little actual use-case for it, (b) it would bloat our SDK code, and (c) there is a simple workaround if needed (spreading into an array `[...realm.subscriptions]`). (The property `length` is available.) Again, something we would love feedback on.
```

Comment:
```suggestion
* We’ve decided to remove numeric indexing and “array methods” from the `SubscriptionSet`, since (a) the team saw little actual use-case for it, (b) it would bloat our SDK code, and (c) there is a simple workaround if needed (spreading into an array `[...realm.subscriptions]`). (The property `length` is available.) Again, something we would love feedback on.
```